thirdeye-server/src/main/java/ai/startree/thirdeye/ThirdEyeServerModule.java

```diff
@@ -69,4 +87,62 @@ public OAuthConfiguration getOAuthConfig(
       AuthConfiguration authConfiguration) {
     return authConfiguration.getOAuthConfig();
   }
+
+  @Singleton
+  @Provides
+  public BasicAuthConfiguration getBasicAuthConfig(
+      AuthConfiguration authConfiguration) {
+    return authConfiguration.getBasicAuthConfig();
+  }
+
+  @Singleton
+  @Provides
+  public AuthFilter getAuthFilters(final OAuthManager oAuthManager) {
+    AuthConfiguration config = configuration.getAuthConfiguration();
+    List<AuthFilter> filters = new ArrayList<>();
+    if(config.isEnabled()) {
+      optional(getBasicAuthFilter()).ifPresent(filters::add);
+      optional(getOAuthFilter(oAuthManager)).ifPresent(filters::add);
+    } else {
+      filters.add(getNoAuthFilter());
+    }
+    if(filters.isEmpty()) {
+      throw new ThirdEyeException(ThirdEyeStatus.ERR_AUTH_FILTERS_DISABLED);
+    }
+    return new ChainedAuthFilter<>(filters);
+  }
+
+  private OAuthCredentialAuthFilter<ThirdEyePrincipal> getNoAuthFilter() {
+    final ThirdEyeAuthenticatorDisabled authenticator = new ThirdEyeAuthenticatorDisabled();
+    return new OAuthCredentialAuthFilter.Builder<ThirdEyePrincipal>()
+        .setAuthenticator(authenticator)
+        .setPrefix(AUTH_BEARER)
+        .buildAuthFilter();
+  }
+
+  private OAuthCredentialAuthFilter<ThirdEyePrincipal> getOAuthFilter(final OAuthManager authManager) {
```

Comment:
Convert this to providers. meaning they should be public with @singleton @provides.

However, they are to be invoked if the auth config is enabled

Reply:
Done